File: backend/internal/dto/email.go
package dto

import "time"

type CreateEmailInboxReq struct {
	Name     string `json:"name"     validate:"required"`
	Provider string `json:"provider" validate:"required,oneof=generic google microsoft"`
	Email    string `json:"email"    validate:"required,email"`

	ImapAddress   string `json:"imap_address"`
	ImapPort      int    `json:"imap_port"`
	ImapLogin     string `json:"imap_login"`
	ImapPassword  string `json:"imap_password"`
	ImapEnableSSL bool   `json:"imap_enable_ssl"`
	ImapEnabled   bool   `json:"imap_enabled"`

	SmtpAddress   string `json:"smtp_address"`
	SmtpPort      int    `json:"smtp_port"`
	SmtpLogin     string `json:"smtp_login"`
	SmtpPassword  string `json:"smtp_password"`
	SmtpEnableSSL bool   `json:"smtp_enable_ssl"`
}

type CreateEmailInboxResp struct {
	InboxResp
	EmailChannelResp
}

type OAuthRedirectResp struct {
	InboxID      int64  `json:"inbox_id"`
	AuthorizeURL string `json:"authorize_url"`
}

type EmailChannelResp struct {
	Email              string    `json:"email"`
	Provider           string    `json:"provider"`
	ImapAddress        *string   `json:"imap_address,omitempty"`
	ImapPort           *int      `json:"imap_port,omitempty"`
	ImapLogin          *string   `json:"imap_login,omitempty"`
	ImapEnableSSL      bool      `json:"imap_enable_ssl"`
	ImapEnabled        bool      `json:"imap_enabled"`
	SmtpAddress        *string   `json:"smtp_address,omitempty"`
	SmtpPort           *int      `json:"smtp_port,omitempty"`
	SmtpLogin          *string   `json:"smtp_login,omitempty"`
	SmtpEnableSSL      bool      `json:"smtp_enable_ssl"`
	VerifiedForSending bool      `json:"verified_for_sending"`
	RequiresReauth     bool      `json:"requires_reauth"`
	EmailCreatedAt     time.Time `json:"email_created_at"`
}

File: frontend/app/schemas/inboxes/email.ts
import { z } from 'zod'

export const emailStepSetupSchema = z.object({
  name: z.string().min(1, 'Name is required'),
  provider: z.enum(['generic', 'google', 'microsoft']),
  email: z.string().email('Valid email is required')
})

export const emailStepImapSchema = z.object({
  imapAddress: z.string().min(1, 'IMAP address is required'),
  imapPort: z.number().int().positive('Port must be positive'),
  imapLogin: z.string().optional(),
  imapPassword: z.string().min(1, 'IMAP password is required'),
  imapEnableSsl: z.boolean()
})

export const emailStepSmtpSchema = z.object({
  smtpAddress: z.string().min(1, 'SMTP address is required'),
  smtpPort: z.number().int().positive('Port must be positive'),
  smtpLogin: z.string().optional(),
  smtpPassword: z.string().min(1, 'SMTP password is required'),
  smtpEnableSsl: z.boolean()
})

export const emailInboxSchema = z.object({
  name: z.string().min(1, 'Name is required'),
  provider: z.enum(['generic', 'google', 'microsoft']),
  email: z.string().email('Valid email is required'),
  imapAddress: z.string().optional(),
  imapPort: z.number().int().positive().optional(),
  imapLogin: z.string().optional(),
  imapPassword: z.string().optional(),
  imapEnableSsl: z.boolean().optional(),
  imapEnabled: z.boolean().optional(),
  smtpAddress: z.string().optional(),
  smtpPort: z.number().int().positive().optional(),
  smtpLogin: z.string().optional(),
  smtpPassword: z.string().optional(),
  smtpEnableSsl: z.boolean().optional()
}).refine(
  data => data.provider !== 'generic' || !!data.imapPassword,
  { message: 'IMAP password is required for generic provider', path: ['imapPassword'] }
)

export type EmailInboxForm = z.infer<typeof emailInboxSchema>
export type EmailStepSetup = z.infer<typeof emailStepSetupSchema>
export type EmailStepImap = z.infer<typeof emailStepImapSchema>
export type EmailStepSmtp = z.infer<typeof emailStepSmtpSchema>
